dao/src/main/java/se/fortnox/reactivewizard/db/transactions/ConnectionScheduler.java

```diff
@@ -15,6 +15,10 @@ public ConnectionScheduler(ConnectionProvider connectionProvider, Scheduler sche
         this.scheduler = scheduler;
     }
 
+    boolean hasConnectionProvider() {
```

Comment:
Since getStatementContextSupplier() returns an Optional, would it be possible to use this fact to avoid setting that statementContextProvider to a value, for the case when there is no connection provider?

Reply:
It is not an optional, it is a supplier. And inside there is the Statement we want to run so I am not so sure about this.